cli/src/helpers/args.rs

```diff
@@ -74,6 +78,49 @@ pub(crate) fn parse_private_key<N: Network>(
     Ok(private_key)
 }
 
+/// Returns the path to the node configuration directory.
+pub(crate) fn parse_node_config_dir(
+    node_config_dir: &Option<Option<PathBuf>>,
+    network: u16,
+    dev: Option<u16>,
+) -> Result<NodeConfigDir> {
+    let new_layout = match node_config_dir {
+        Some(Some(custom)) => return Ok(NodeConfigDir::new(custom.clone())),
+        Some(None) => true,
```

Comment:
What is this, when someone passes `--node-config-dir` without any arguments? I think allowing for this will be confusing.

Reply:
It allows you to enable the new layout without specifying a custom directory. I can maybe add a new flag instead? That one would also be removed in the release after the next or so, when stop supporting the layout.

Reply:
I would not support the legacy format. Instead:
- when passing the --node-config-dir <path> it will be set to the path. It is mandatory for validators to set and therefore think about migration. For clients the folder contents are not that critical and nobody will care in the near-term.
- if you don't use the flag, use a hardcoded default path.